unified-scheduler-pool/src/lib.rs

```diff
@@ -3035,14 +3197,58 @@ mod tests {
             .schedule_execution(tx1, BLOCKED_TRANSACTION_INDEX)
             .unwrap();
 
+        let bank = BankWithScheduler::new(bank, Some(scheduler));
+        poh_recorder
+            .write()
+            .unwrap()
+            .set_bank(bank.clone_with_scheduler(), true);
+
         // Wait a bit for the scheduler thread to decide to block tx1
-        std::thread::sleep(std::time::Duration::from_secs(1));
+        std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
sorry for being persistent on all these timings in tests - I just view every sleep as a potential for a flaky test.

Wdyt of the following?

We are basically waiting here for the scheduler thread to pick up things from the channel? Can we add a (test-only) fn on the scheduler that just returns the channel length? that way we can know whether or not the task has been recv'd or not, deterministically.
it doesn't fully solve the issue, since it may have been picked up from the channel, but not yet processed.
a possibility here is to just have the scheduler & tests share an atomic that is # of tasks processed by scheduler - wait for that to increment, and you know you can continue.

Reply:
well, I'll appreciate re-raising your concern here. fortunately, i came up with some neat testing impl from your suggestion: 61d9465ee248910f053150050c1b0c68962ebab4

this test is migrated to `sleepless_testing` and all new & pre-existing sleeps has gone now.